packages/cli/src/ActiveWorkflowRunner.ts

```diff
@@ -333,20 +337,30 @@ export class ActiveWorkflowRunner {
 	 * @returns {string[]}
 	 * @memberof ActiveWorkflowRunner
 	 */
-	async getActiveWorkflows(userId?: string): Promise<IWorkflowDb[]> {
-		// TODO UM: make userId mandatory?
-		const queryBuilder = Db.collections.Workflow!.createQueryBuilder('w');
-		queryBuilder.andWhere('active = :active', { active: true });
-		queryBuilder.select('w.id');
-		if (userId) {
-			queryBuilder.innerJoin('w.shared', 'shared');
-			queryBuilder.andWhere('shared.user = :userId', { userId });
+	async getActiveWorkflows(user?: User): Promise<IWorkflowDb[]> {
+		let activeWorkflows: WorkflowEntity[] = [];
+
+		if (!user || user.globalRole.name === 'owner') {
+			activeWorkflows = await Db.collections.Workflow!.find({
+				select: ['id'],
+				where: { active: true },
+			});
+		} else {
+			const shared = await Db.collections.SharedWorkflow!.find({
+				relations: ['workflow'],
+				where: whereClause({
```

Comment:
I think it's better to have possibly additional filters to this whereClause allowing us to use `active = true` as part of it. This would reduce the need for the accumulator below without sacrificing re-usability.

I know the `whereClause` function already expects a third optional argument so maybe we should rethink how this can work.

Let me know if you think we should change this now or merge first and then make all changes at once when refactoring code to centralize these helpers into a model-like class.

Reply:
Agree, let's do away with `reduce()` and keep the filters flexible, ideally merging first and then refactoring, so that we have a full view of all the call variants that we need to account for in `UserManagementModel`.